module/mfrc522_spi.c

```diff
@@ -17,22 +17,128 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
 	return byte;
 }
 
-u8 mfrc522_get_version(void)
+int mfrc522_get_version(void)
 {
-	return mfrc522_register_read(mfrc522_spi, MFRC522_VERSION_REG);
+	u8 version;
+	int ret;
+
+	ret = mfrc522_register_read(mfrc522_spi, MFRC522_VERSION_REG, &version,
+				    1);
+
+	if (ret < 0)
+		return ret;
+
+	return version;
+}
+
+void mfrc522_fifo_flush(void)
+{
+	u8 flush_byte = 1 << 7;
+
+	mfrc522_register_write(mfrc522_spi, MFRC522_FIFO_LEVEL_REG, flush_byte);
+}
+
+static void wait_for_cmd(void)
+{
+	int cmd;
+
+	do {
+		cmd = mfrc522_read_command();
+		pr_info("[MFRC522] Current command is: 0x%x\n", cmd);
+	} while (cmd != MFRC522_COMMAND_IDLE);
 }
 
-u8 mfrc522_register_read(struct spi_device *client, u8 reg)
+void mfrc522_send_command(u8 rcv_off, u8 power_down, u8 command)
 {
+	u8 command_byte = rcv_off << 5 | power_down << 4 | command;
```

Comment:
```c
#define XXXXX 5
#define YYYYY 4
```

Reply:
That's not fair that was YOUR code :P But yeah I agree that we should macros instead